external/loaders/loaders/batches/_sequences.py

```diff
@@ -69,13 +69,20 @@ def files(self):
 
     @classmethod
     def dump(cls, dataset, path):
-        joblib.dump(dataset, path)
+        try:
+            joblib.dump(dataset.load(), path)
+        except AttributeError:
+            joblib.dump(dataset, path)
 
     def __len__(self):
         return len(self.files)
 
     def __getitem__(self, i):
-        return joblib.load(self.files[i])
+        slice_value = self.files[i]
```

Comment:
Is there a generic for containers that allow slices?

Reply:
I assumed it was Sequence because all the built-in sequences accept slices. The docs say "integer" but there's no type hint on the definition of Sequence.

I think we're pretty safe from this getting relied on in our package code (when would we slice a batches sequence? the idea is that batch is the largest we can process at once), but it will make things easier when I'm working in notebooks.